### DISCUSSION TITLE
A simple way to play the selection looped (without needing 3 key presses!)

### DISCUSSION THREAD

**Discussion by Mr-Personality** [audacity/audacity#7195] [Ideas & Feature Requests]
### Your idea

**Problem:** Like many sound editors, I use looping extensively to hear what I'm working on, so loop controls really do matter to many people.

Originally, "Loop play" was a very simple function to use, requiring **only _one_ key press**: You just hit the "loop play" key (instead of space) to play the selection looped.

More recently, this simple one-key function has been replaced with quite a complicated system. If you want to quickly play the selection looped, instead of hitting a single key, you now must hit **_three keys:_**

1. First, you need to hit “Set Loop To Selection”
2. Then you need to hit space to play
3. Then after you've finished listening to the loop (which may be only a split-second later), you have to turn looping off in order to 'rescue' the playhead from being trapped in the selection area.

So not only does this new system require 3 times the key presses, it's also far more complicated, less intuitive, and can be confusing, with multiple different settings working together. First, you have the control to set the loop to the selection. Then you have the loop on/off control (and you need to always keep in mind what mode you're in). And then you can also control the start and end points of the loop, which are represented with special markers. Just writing this gets me confused!

Granted, this new system allows users to have a fixed loop area, with custom markers, which never changes while you're working on it. However, I really don't think many people ever need to work this way. I think the majority of users who want to loop won't need anything more complex than to hear the selection looping.

My solution:

**Solution 1**: A simple solution would be to simply bring back the "Loop Play" button (in addition to the existing controls). However, I think there's a far better solution:

**Solution 2**:
• Make "Loop" on by default
• Add a new setting (also enabled by default) called _"Loop selection instead of using loop markers"_

When this setting is enabled ☑ (and Loop is on), the loop markers will be completely ignored and hidden, and it will simply loop the selection

When this setting is disabled ☐ (and Loop is on), the loop will follow the loop markers.

The advantage of this system is that it makes looping easy for everyone to use - in fact, it's EVEN EASIER than it used to be with the "Loop Play" key! Because now users don't even need a "Loop Play" key - the space bar does EVERYTHING.

If there's no selection, hitting space will simply play. But if there's a selection (and Loop is on) then the selection will play it looped. Anyone wanting more ambitious looping controls will be able to go into the Loop settings and disable this new setting, to get more control. But for most users, this will be ideal, and extremely simple and intuitive to use. When they first run Audacity, playing a selection will always loop play.

### Problem to be solved

See above (I don't want to write the solution before the problem, that to me is backwards).

**Comment by mkj42** [comment]
I also recently updated Audacity and disliked the disappearance of the old way.
In 2.1.3 it worked where holding shift changed both play buttons into looping icons, and shift+either play button = loop the selection.
Now shift+either play button toggles recording (!).
Please consider bringing back the old functionality.

Side note:
Could anyone say when this feature was changed between 2.1.3 and 3.6.4?
I might consider rolling back just to have it the old way.
